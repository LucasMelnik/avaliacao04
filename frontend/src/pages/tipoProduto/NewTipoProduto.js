import React, { useState } from "react";
import api from '../../api/index'

export default function NewTipoProduto() {
    const [tipoProduto, setTipoProduto] = useState({})

    function handleChangeNome(props) {
        setTipoProduto({...tipoProduto,
            nome: props.target.value
        })
    } 

    function handleSubmit(event) {
        api.post('/tipoproduto', tipoProduto)
    }

    return(
        <form onSubmit={handleSubmit}>
            <div>
                <label>Nome</label>
                <input id="nome" type="text" value={tipoProduto.nome} onChange={handleChangeNome}/>
            </div>
            <button id="submit" type="submit">Enviar</button>
        </form>
    )
}