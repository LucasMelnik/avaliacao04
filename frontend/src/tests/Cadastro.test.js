import React from "react"
import { shallow, } from 'enzyme'
import '@testing-library/jest-dom/extend-expect'
import '@testing-library/jest-dom'
import NewTipoProduto from '../pages/tipoProduto/index'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() });

describe("Testing NewTipoProduto Component", () => {
    it("should render correctly", () => {
        const wrapper = shallow(
            <NewTipoProduto />
        )

        expect(wrapper).toMatchSnapshot();
    })
    
})