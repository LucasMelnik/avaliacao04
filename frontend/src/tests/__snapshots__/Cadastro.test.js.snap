exports[`Testing NewTipoProduto Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TipoProduto />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          to="/tipo-produto/new">
          +
</Unknown>,
        <table>
          <thead>
                    <tr>
                              <th
                                        style={
                                                  Object {
                                                            "border": "solid 1px",
                                                          }
                                        }>
                                        ID
                              </th>
                              <th
                                        style={
                                                  Object {
                                                            "border": "solid 1px",
                                                          }
                                        }>
                                        Nome
                              </th>
                              <th
                                        style={
                                                  Object {
                                                            "border": "solid 1px",
                                                          }
                                        }>
                                        Ação
                              </th>
                    </tr>
          </thead>
          <tbody>
                    
          </tbody>
</table>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "+",
          "to": "/tipo-produto/new",
        },
        "ref": null,
        "rendered": "+",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                            <th
                                          style={
                                                        Object {
                                                                      "border": "solid 1px",
                                                                    }
                                          }>
                                          ID
                            </th>
                            <th
                                          style={
                                                        Object {
                                                                      "border": "solid 1px",
                                                                    }
                                          }>
                                          Nome
                            </th>
                            <th
                                          style={
                                                        Object {
                                                                      "border": "solid 1px",
                                                                    }
                                          }>
                                          Ação
                            </th>
              </tr>
</thead>,
            <tbody>
              
</tbody>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th
                                style={
                                                Object {
                                                                "border": "solid 1px",
                                                              }
                                }>
                                ID
                </th>
                <th
                                style={
                                                Object {
                                                                "border": "solid 1px",
                                                              }
                                }>
                                Nome
                </th>
                <th
                                style={
                                                Object {
                                                                "border": "solid 1px",
                                                              }
                                }>
                                Ação
                </th>
</tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    style={
                                        Object {
                                                            "border": "solid 1px",
                                                          }
                    }>
                    ID
</th>,
                  <th
                    style={
                                        Object {
                                                            "border": "solid 1px",
                                                          }
                    }>
                    Nome
</th>,
                  <th
                    style={
                                        Object {
                                                            "border": "solid 1px",
                                                          }
                    }>
                    Ação
</th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ID",
                    "style": Object {
                      "border": "solid 1px",
                    },
                  },
                  "ref": null,
                  "rendered": "ID",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Nome",
                    "style": Object {
                      "border": "solid 1px",
                    },
                  },
                  "ref": null,
                  "rendered": "Nome",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Ação",
                    "style": Object {
                      "border": "solid 1px",
                    },
                  },
                  "ref": null,
                  "rendered": "Ação",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            to="/tipo-produto/new">
            +
</Unknown>,
          <table>
            <thead>
                        <tr>
                                    <th
                                                style={
                                                            Object {
                                                                        "border": "solid 1px",
                                                                      }
                                                }>
                                                ID
                                    </th>
                                    <th
                                                style={
                                                            Object {
                                                                        "border": "solid 1px",
                                                                      }
                                                }>
                                                Nome
                                    </th>
                                    <th
                                                style={
                                                            Object {
                                                                        "border": "solid 1px",
                                                                      }
                                                }>
                                                Ação
                                    </th>
                        </tr>
            </thead>
            <tbody>
                        
            </tbody>
</table>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "+",
            "to": "/tipo-produto/new",
          },
          "ref": null,
          "rendered": "+",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                                <th
                                                style={
                                                                Object {
                                                                                "border": "solid 1px",
                                                                              }
                                                }>
                                                ID
                                </th>
                                <th
                                                style={
                                                                Object {
                                                                                "border": "solid 1px",
                                                                              }
                                                }>
                                                Nome
                                </th>
                                <th
                                                style={
                                                                Object {
                                                                                "border": "solid 1px",
                                                                              }
                                                }>
                                                Ação
                                </th>
                </tr>
</thead>,
              <tbody>
                
</tbody>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th
                                    style={
                                                      Object {
                                                                        "border": "solid 1px",
                                                                      }
                                    }>
                                    ID
                  </th>
                  <th
                                    style={
                                                      Object {
                                                                        "border": "solid 1px",
                                                                      }
                                    }>
                                    Nome
                  </th>
                  <th
                                    style={
                                                      Object {
                                                                        "border": "solid 1px",
                                                                      }
                                    }>
                                    Ação
                  </th>
</tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th
                      style={
                                            Object {
                                                                  "border": "solid 1px",
                                                                }
                      }>
                      ID
</th>,
                    <th
                      style={
                                            Object {
                                                                  "border": "solid 1px",
                                                                }
                      }>
                      Nome
</th>,
                    <th
                      style={
                                            Object {
                                                                  "border": "solid 1px",
                                                                }
                      }>
                      Ação
</th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "ID",
                      "style": Object {
                        "border": "solid 1px",
                      },
                    },
                    "ref": null,
                    "rendered": "ID",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Nome",
                      "style": Object {
                        "border": "solid 1px",
                      },
                    },
                    "ref": null,
                    "rendered": "Nome",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Ação",
                      "style": Object {
                        "border": "solid 1px",
                      },
                    },
                    "ref": null,
                    "rendered": "Ação",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
